src/main/scala/io/github/pauljamescleary/petstore/domain/users/UserService.scala

```diff
@@ -1,42 +1,39 @@
 package io.github.pauljamescleary.petstore.domain.users
 
-import scala.language.higherKinds
-
 import cats._
 import cats.data._
 import io.github.pauljamescleary.petstore.domain.{UserAlreadyExistsError, UserNotFoundError}
 
-class UserService[F[_]](userRepo: UserRepositoryAlgebra[F], validation: UserValidationAlgebra[F]) {
-  import cats.syntax.all._
+class UserService[F[_]: Monad](userRepo: UserRepositoryAlgebra[F], validation: UserValidationAlgebra[F]) {
 
-  def createUser(user: User)(implicit M: Monad[F]): EitherT[F, UserAlreadyExistsError, User] =
+  def createUser(user: User): EitherT[F, UserAlreadyExistsError, User] =
     for {
       _ <- validation.doesNotExist(user)
-      saved <- EitherT.liftF(userRepo.create(user))
+      saved <- EitherT.liftF(userRepo.put(user))
     } yield saved
 
-  def getUser(userId: Long)(implicit M: Monad[F]): EitherT[F, UserNotFoundError.type, User] =
-    EitherT.fromOptionF(userRepo.get(userId), UserNotFoundError)
+  def getUser(userId: Long): EitherT[F, UserNotFoundError.type, User] =
+    EitherT.fromOptionF(userRepo.get(userId).value, UserNotFoundError)
```

Comment:
is there any way we can more simply go from `OptionT` to `EitherT` without peeking into `.value`?

Maybe it is just me, but `.value` tends to bother me.

Reply:
I think your idea of coming to a consensus implementation strategy for this in #82 is good. I've tried to revert all these fudgy changes for now.